File: Data Structures & Algorithms/same-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        return compare_dfs(p,q);
    }
    private boolean compare_dfs(TreeNode p , TreeNode q ){
        if( p == null && q == null ){
            return true ;
        }
        if(p == null || q == null ){
            return false ;
        }
        if(p.val!=q.val){
            return false;
        }
        return compare_dfs(p.left , q.left) && compare_dfs(p.right , q.right);
    }
}

File: Data Structures & Algorithms/house-robber-iii/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rob(self, root: Optional[TreeNode]) -> int:
        def dfs(node):
            if node is None:
                return [0,0]
            left = dfs(node.left)
            right = dfs(node.right)

            # don't rob the current node
            not_rob = max(left) + max(right)

            rob = node.val + left[0] + right[0]

            return [not_rob , rob]
        result = dfs(root)
        return max(result )



        